src/skmultiflow/utils/data_structures.py

```diff
@@ -1014,3 +1015,85 @@ def get_info(self):
                ' - n_samples: ' + str(self.n_samples) + \
                ' - max_size: ' + str(self.max_size) + \
                ' - dtype: ' + str(self.dtype)
+
+class TimeManager(object):
+
+    _columns = ["X", "y_real", "y_pred", "arrival_time", "available_time"]
+
+    def __init__(self, timestamp):
+        # get initial timestamp
+        self.timestamp = timestamp
+        # create dataframe to save time data
+        # X = features
+        # y_real = real label
+        # y_pred = predicted label for each model being evaluated
+        # arrival_time = arrival timestamp of the sample
+        # available_time = timestamp when the sample label will be avilable
+        self.queue = pd.DataFrame(columns=self._columns)
```

Comment:
I'd use `numpy` array instead of Pandas DataFrame. It consumes less memory and Pandas don't make it simple in this case. 
[Numpy Vs Pandas Performance Comparison](http://gouthamanbalaraman.com/blog/numpy-vs-pandas-comparison.html)

Reply:
You already mentioned it. Sorry I missed it in the description!